### PR TITLE
Migrate testing to Jest

### PR DESCRIPTION
**Pull request by adonesky1**
Migrates test suite from Mocha to Jest for better overall tooling, including coverage reports!

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module.exports = {
```
Comment (mcmire) [inline jest.config.js:1]: Also of note, we can eventually replace this with https://github.com/MetaMask/metamask-module-template/blob/main/jest.config.ts, but that can happen when this repo gets converted to TypeScript, which @Gudahtt is working on.

**Comment by adonesky1** [comment]
> This seems to remove the old Mocha tests but not convert them to Jest.

Not sure I follow? The tests are running with the Jest test runner now?

**Comment by mcmire** [comment]
@adonesky1 You removed `test/index.js`, so there aren't tests anymore. I think we'd want to convert those tests to Jest, right?

**Comment by adonesky1** [comment]
🤔  @mcmire & @Gudahtt why am I seeing a different diff than you seem to be?
Screen Shot 2022-04-14 at 2 48 18 PM

**Review by mcmire — APPROVED** [review]
My bad! I don't know why I thought that `test/index.js` was deleted. Looks good!

**Comment by Gudahtt** [comment]
Sorry it was probably the line deletion count :sweat_smile: It was collapsed in the diff view by default and I thought at a glance that it was a removed file.

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,6 @@ package-lock.json
 
 # Optional eslint cache
 .eslintcache
+
+# Jest
+/coverage
```
Comment (Gudahtt) [inline .gitignore:8]: Nit: This should end in a newline. Usually the editorconfig file ensures this happens automatically in supported editors, but we haven't finished standardizing this repo yet.

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: .gitignore (hunk 2 above)
Comment (Gudahtt) [reply .gitignore:8]: I can fix this later, not a big deal

**Pull request merged**